Adding SQS Message ID as a header to SqsReceiverMessage

austin0150 opened the pull request:
## Description
Adding the SQS Message ID as a header in the SqsReceiverMessage.
Currently the ID of the SQS message being read is not accessible.


## Type of change: 
3. New feature (non-breaking change that adds functionality)

## Checklist:

- Have you reviewed your own code? Do you understand every change?
- Are you following the [contributing guidelines](../blob/main/CONTRIBUTING.md)?
- Have you added tests that prove your fix is effective or that this feature works?
- New and existing unit tests pass locally with these changes?
- Have you made corresponding changes to the documentation?
- Will this change require an update to an example project? (if so, create an issue and link to it)

---

<sup>_[Reviewer guidelines](../blob/main/CONTRIBUTING.md#reviewing-changes)_</sup>

NeilSchiefer commented on Tests/RockLib.Messaging.SQS.Tests/SQSTests.cs:
```diff
@@ -412,6 +412,56 @@ public static void SQSReceiverMessageWithUnpackSNSTrueCorrectlyUnpacksSNSMessage
+            var sqsReceiverMessage = new SQSReceiverMessage(message, c => Task.FromResult(0), c => Task.FromResult(0), unpackSNS: true);
+
+
+            //var messageId = 
```
Can just remove this comment.

austin0150 commented:
Whoops, sorry about those.
Removed the comment and updated the changelog

eslutz commented on Tests/RockLib.Messaging.SQS.Tests/SQSTests.cs:
```diff
@@ -412,6 +412,54 @@ public static void SQSReceiverMessageWithUnpackSNSTrueCorrectlyUnpacksSNSMessage
+                Body = @"{""TestMessageItem"":""ThisIsAFakeItem""}"
+            };
+
+            var sqsReceiverMessage = new SQSReceiverMessage(message, c => Task.FromResult(0), c => Task.FromResult(0), unpackSNS: false);
+
+            var messageId = sqsReceiverMessage.Headers.GetValue<string>("SQS.MessageID");
+
+            messageId.Should().NotBeNull();
+
+            sqsReceiverMessage.Dispose();
```
Can you update line 424 to `using var sqsReceiverMessage` and then you can remove the `sqsReceiverMessage.Dispose()` on line 430.  That way it's consistent with the other tests in the file.

eslutz commented on Tests/RockLib.Messaging.SQS.Tests/SQSTests.cs:
```diff
@@ -412,6 +412,54 @@ public static void SQSReceiverMessageWithUnpackSNSTrueCorrectlyUnpacksSNSMessage
+                Body = snsMessage
+            };
+
+            var sqsReceiverMessage = new SQSReceiverMessage(message, c => Task.FromResult(0), c => Task.FromResult(0), unpackSNS: true);
+ 
+            Assert.Throws<KeyNotFoundException>(() => sqsReceiverMessage.Headers.GetValue<string>("SQS.MessageID"));
+
+            sqsReceiverMessage.Dispose();
```
Same here.  Add the `using` statement and get rid of the `sqsReceiverMessage.Dispose()`.

NeilSchiefer commented on RockLib.Messaging.SQS/CHANGELOG.md:
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 	
 #### Added
 - Added `.editorconfig` and `Directory.Build.props` files to ensure consistency.
+- Added "SQS.MessageID" as header to SQSReceiverMessages when the not attempting to unpack as an SNS message
```
Probably can remove the 'the' or reword.

eslutz commented on RockLib.Messaging.SQS/CHANGELOG.md:
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 	
 #### Added
 - Added `.editorconfig` and `Directory.Build.props` files to ensure consistency.
+- Added "SQS.MessageID" as header to SQSReceiverMessages when the not attempting to unpack as an SNS message
```
A couple issues at the end of the line with `unpack as an SNS message`.  Can you update it to `unpack as a SQS message.`.

NeilSchiefer replied:
Thats actually correct the code is supports to content of the sqs message as an SNS message I guess.  https://github.com/RockLib/RockLib.Messaging/blob/main/RockLib.Messaging.SQS/SQSReceiverMessage.cs#L51

eslutz replied:
I noticed I had a mistake in my correction, and you caught it. Nice!

austin0150 commented:
Updated to use "Using" instead of "Dispose" that's much cleaner, thanks. I was taking the "use dispose" editor alert too literally. 
Updated my wording in the changelog as well.

eslutz reviewed (approved):
Nice work!  Thanks for the addition 👍🏻

kristycurrier commented on Tests/RockLib.Messaging.SQS.Tests/SQSTests.cs:
```diff
@@ -412,6 +412,52 @@ public static void SQSReceiverMessageWithUnpackSNSTrueCorrectlyUnpacksSNSMessage
+        {
+
+            var snsMessage = @"{
+  ""Type"" : ""Notification"",
```
Please indent this along with the other 2.

NeilSchiefer reviewed (approved):
👍
